tests/backup_test.php

```diff
@@ -0,0 +1,226 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace qtype_crossword;
+
+use qtype_crossword;
+
+defined('MOODLE_INTERNAL') || die();
+global $CFG;
+require_once($CFG->libdir . "/phpunit/classes/restore_date_testcase.php");
+require_once($CFG->dirroot . '/question/type/crossword/questiontype.php');
+
+/**
+ * Unit tests for backup/restore process in crossword qtype.
+ *
+ * @package qtype_crossword
+ * @copyright 2023 The Open University
+ * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class backup_test extends \restore_date_testcase {
+
+    /**
+     * Load required libraries
+     */
+    public static function setUpBeforeClass(): void {
+        global $CFG;
+        require_once("{$CFG->dirroot}/backup/util/includes/restore_includes.php");
+    }
+
+    /**
+     * Restore the crossword backup file in the fixture folder base on filemame.
+     *
+     * @param string $filename backup file name in the fixture folder.
+     * @param string $coursefullname course full name to be restored.
+     * @param string $courseshortname course short name to be restored.
+     * @return mixed bool|stdClass return the question object restored.
+     */
+    protected function restore_crossword_question_backup_file_to_course_shortname(string $filename, string $coursefullname,
+        string $courseshortname) {
+        global $DB, $USER;
+        $testfixture = __DIR__ . '/fixtures/' . $filename;
+
+        // Extract our test fixture, ready to be restored.
+        $backuptempdir = 'qtype_crossword';
+        $backuppath = make_backup_temp_directory($backuptempdir);
+        get_file_packer('application/vnd.moodle.backup')->extract_to_pathname($testfixture, $backuppath);
+        // Do the restore to new course with default settings.
+        $categoryid = $DB->get_field('course_categories', 'MIN(id)', []);
+        $courseid = \restore_dbops::create_new_course($coursefullname, $courseshortname, $categoryid);
+
+        $controller = new \restore_controller($backuptempdir, $courseid, \backup::INTERACTIVE_NO, \backup::MODE_GENERAL, $USER->id,
+            \backup::TARGET_NEW_COURSE);
+
+        $controller->execute_precheck();
+        $controller->execute_plan();
+        $controller->destroy();
+
+        return $DB->get_record('question', []);
```

Comment:
You need to read https://tracker.moodle.org/browse/MDL-62483

This does not work in CodeRunner (or similar) is installed.